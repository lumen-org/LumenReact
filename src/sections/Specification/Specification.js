import React, { Component } from "react";
import TitelH1 from "../../components/Titles/TitelH1";
import Field from "../../components/Field";
import TitelH2 from "../../components/Titles/TitleH2";
import "./Specification.css"

class Specification extends Component {

  specification_values() {
    return Object.entries(this.props.data_lists).map(([title, data]) =>
      <Field title={title} data={data} dropable={true}/>
    );
  }

  render() {
    return (
      <div className="specification">
        <TitelH1 value={"Schema"}/>
        {this.specification_values()}
        <TitelH2 value={"Drop here to remove"}/>
      </div>
    );
  }
}

export default Specification;
