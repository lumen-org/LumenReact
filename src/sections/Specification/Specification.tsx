import React, { Component } from "react";
import PropTypes from "prop-types";
import TitleH1 from "../../components/Titles/TitleH1";
import TitleH2 from "../../components/Titles/TitleH2";
import Field from "../../components/Field";
import Facet from "../Facet";
import "./Specification.css";

type SpecificationProps = {
  specifications: object,
  facets: object
}

class Specification extends Component<SpecificationProps> {
  render() {
    const { specifications, facets } = this.props;
    return (
      <div className="specification">
        <TitleH1 value={"Spezification"}/>
        {Object.entries(specifications).map((item, index) => (
          <Field
            title={item[0]}
            data={item[1]}
            dropable={true}
          />
        ))}
        <TitleH2 value={"Drop here to remove"}/>
        <TitleH1 value={"Facets"}/>
        <Facet text="Test" facets={facets}/>
      </div>
    );
  }
}

export default Specification;
