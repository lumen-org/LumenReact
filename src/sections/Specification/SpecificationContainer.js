import React from "react";
import Specification from "./Specification";
import { connect } from "react-redux";
import { selectActiveSpecificationId } from "../../states/visualizations/selector";

class SpecificationContainer extends React.Component {
  render() {
    const specifications = this.props.specifications.byId;
    return (<div>
        {
          this.props.activeSpecification !== -1 &&
          <Specification
            specifications={specifications[this.props.activeSpecification].specification}
            facets={specifications[this.props.activeSpecification].facets}
          />
        }
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    specifications: state.specifications.specifications,
    activeSpecification: selectActiveSpecificationId(state)
  };
};

export default connect(mapStateToProps, null)(SpecificationContainer);
