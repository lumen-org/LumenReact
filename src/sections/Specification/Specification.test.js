import React from "react";
import ReactDOM from "react-dom";
import Specification from "./Specification";
import TestRenderer, { act } from "react-test-renderer";

// https://jestjs.io/docs/en/tutorial-react
/// what to mock ? Field, TitelH1, TitelH2, , Facet

jest.mock("../../components/Field", () => (props) => (<div>
    <div>
        {props.title}
    </div>
    <div>
        {props.data}
    </div>
</div>));

jest.mock("../../components/Titles/TitleH1", () => (props) => (
    <div>
        {props.value}
    </div>
));

jest.mock("../../components/Titles/TitleH2", () => (props) => (
    <h2>
        {props.value}
    </h2>
));

jest.mock("../Facet", () => (props) => (
    <div>
        {props.facets}
    </div>
))

test("Schema displays dimensions correctly", () => {
    const specifications = { "X_Axis": [], "Y-Axis": []};
    const facets = "bla";//{ "Prediction": {}, "Data Points": {}, "Marginals": {}, "Desity": {} };

    const component = TestRenderer.create(
        <Specification specifications={specifications} facets={facets} />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

test("Specification throws no exception when all attributes are passed correctly", () => {
    const specifications = { "X_Axis": [], "Y-Axis": []};
    const facets = "bla";//{ "Prediction": {}, "Data Points": {}, "Marginals": {}, "Desity": {} };

    jest.spyOn(console, "error")

    TestRenderer.create(
        <Specification specifications={specifications} facets={facets} />
    )
    expect(console.error).not.toBeCalled()
})

/*

test("Specification throws exception when no attribute is passed", () => {

    jest.spyOn(console, "error");
    act(() => {
        TestRenderer.create(
            <Specification />
        );
    });
    expect(console.error).toBeCalled();
})


test("Schema throws exception when no quantitative attribute is passed", () => {
    const categorical = ["sepal_length", "sepal_width", "petal_length", "petal_width"]

    jest.spyOn(console, "error")

    TestRenderer.create(
        <Specification categorical={categorical} />
    )
    expect(console.error).toBeCalled()
})

test("Schema throws exception when no categorical attribute is passed", () => {
    const quantitative = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", "15", "16", "species"]

    jest.spyOn(console, "error")
    TestRenderer.create(
        <Specification quantitative={quantitative} />
    )

    expect(console.error).toBeCalled()
})




// My proposel evtl.
// beforeEach(() => {
// jest.spyOn(console, 'error')
// jest.spyOn(console, 'warn')
// })
// afterEach(() => {
// /* eslint-disable no-console,jest/no-standalone-expect */
// expect(console.error).not.toBeCalled()
// expect(console.warn).not.toBeCalled()
// })
