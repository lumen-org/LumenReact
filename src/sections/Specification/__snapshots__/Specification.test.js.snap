// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema displays dimensions correctly 1`] = `
<div
  className="specification"
>
  <div>
    Spezification
  </div>
  <div>
    <div>
      X_Axis
    </div>
    <div />
  </div>
  <div>
    <div>
      Y-Axis
    </div>
    <div />
  </div>
  <h2>
    Drop here to remove
  </h2>
  <div>
    Facets
  </div>
  <div>
    bla
  </div>
</div>
`;
